api/src/test/java/org/openmrs/api/db/hibernate/HibernatePatientDAOTest.java

```diff
@@ -63,4 +70,116 @@ public void getPatientIdentifiers_shouldGetByPatients() {
+		person2.addName(new PersonName("Ioan1", "Theo1", "Fletcher1"));
+		person2 = hibernatePersonDAO.savePerson(person2);
+		hibernatePatientDao.savePatient(new Patient(person2));
+
+		// when
+		List<Patient> duplicatePatients = hibernatePatientDao
+			.getDuplicatePatientsByAttributes(Arrays.asList("givenName", "middleName", "familyName"));
+
+		// then
+		assertThat(duplicatePatients.size(), equalTo(0));
+	}
+
+	@Test
+	public void getDuplicatePatientsByAttributes_shouldReturnPatientsWithDuplicatedAllNames() {
+		// given
+		Person person1 = new Person();
+		person1.addName(new PersonName("Ioan", "Theo", "Fletcher"));
+		person1 = hibernatePersonDAO.savePerson(person1);
+		hibernatePatientDao.savePatient(new Patient(person1));
+
+		Person person2 = new Person();
+		person2.addName(new PersonName("Ioan", "Theo", "Fletcher"));
+		person2 = hibernatePersonDAO.savePerson(person2);
+		hibernatePatientDao.savePatient(new Patient(person2));
+
+		// when
+		List<Patient> duplicatePatients = hibernatePatientDao
+			.getDuplicatePatientsByAttributes(Arrays.asList("givenName", "middleName", "familyName"));
+		
+		// then
+		assertThat(duplicatePatients.size(), equalTo(2));
+	}
+
+	@Test
+	public void getDuplicatePatientsByAttributes_shouldReturnPatientsWithDuplicatedFamilyNames() {
+		// given
+		Person person1 = new Person();
+		person1.addName(new PersonName("Ioan", "Theo", "Fletcher"));
+		person1 = hibernatePersonDAO.savePerson(person1);
+		hibernatePatientDao.savePatient(new Patient(person1));
+
+		Person person2 = new Person();
+		person2.addName(new PersonName("Ioan1", "Theo1", "Fletcher"));
+		person2 = hibernatePersonDAO.savePerson(person2);
+		hibernatePatientDao.savePatient(new Patient(person2));
+
+		// when
+		List<Patient> duplicatePatients = hibernatePatientDao
+			.getDuplicatePatientsByAttributes(singletonList("familyName"));
+
+		// then
+		assertThat(duplicatePatients.size(), equalTo(2));
+	}
+
+	@Test
+	public void getDuplicatePatientsByAttributes_shouldReturnPatientsWithDuplicatedAllFields() {
+		// given
+		Person person1 = new Person();
+		person1.addName(new PersonName("Ioan", "Theo", "Fletcher"));
+		person1.setGender("M");
+		person1.setBirthdate(Date.valueOf("2021-06-26"));
+		person1 = hibernatePersonDAO.savePerson(person1);
+		hibernatePatientDao.savePatient(new Patient(person1));
+
+		Person person2 = new Person();
+		person2.addName(new PersonName("Ioan", "Theo", "Fletcher"));
+		person2.setGender("M");
+		person2.setBirthdate(Date.valueOf("2021-06-26"));
+		person2 = hibernatePersonDAO.savePerson(person2);
+		hibernatePatientDao.savePatient(new Patient(person2));
+
+		// when
+		List<Patient> duplicatePatients = hibernatePatientDao
+			.getDuplicatePatientsByAttributes(Arrays.asList("gender", "birthdate", "givenName", "middleName", "familyName"));
+
+		// then
+		assertThat(duplicatePatients.size(), equalTo(2));
+	}
+
+	@Test
+	public void getDuplicatePatientsByAttributes_shouldReturnPatientsWithDuplicatedAllButOneFields() {
```

Comment:
@BartlomiejRasztabiga  could this need a better wording, failing t o understand it

Reply:
Sure I will find a better one. I wanted it to mean that those two patients have all duplicated fields besides gender :)

Reply:
I've updated this test's name in the latest commit